### PR TITLE
RHIDP-4201: docs(homepage): add techdocs

### PR DESCRIPTION
**Pull request by jerolimov**
## Description

Add some tech docs for the dynamic home page plugin.

## Which issue(s) does this PR fix

- Fixes https://issues.redhat.com/browse/RHIDP-4201

## PR acceptance criteria

Please make sure that the following steps are complete:

- [ ] GitHub Actions are completed and successful
- [ ] Unit Tests are updated and passing
- [ ] E2E Tests are updated and passing
- [ ] Documentation is updated if necessary (requirement for new features)
- [ ] Add a screenshot if the change is UX/UI related

### CODE REVIEW

File: plugins/dynamic-home-page/catalog-info.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  subcomponentOf: janus-idp-backstage-plugins
+---
+# https://backstage.io/docs/features/software-catalog/descriptor-format#kind-component
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  name: janus-idp-dynamic-home-page-frontend
+  title: '@janus-idp/backstage-plugin-dynamic-home-page'
+  description: Home Page plugin based on the upstream home [page] plugin that can be extended via (dynamic) plugins.
+  annotations:
```
Comment (invincibleJai) [inline plugins/dynamic-home-page/catalog-info.yaml:38]: how is this different from the above component?

File: plugins/dynamic-home-page/catalog-info.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  title: Dynamic Home Page plugin
+  description: Home Page plugin based on the upstream home [page] plugin that can be extended via (dynamic) plugins.
+  annotations:
+    backstage.io/source-location: url:https://github.com/janus-idp/backstage-showcase/tree/main/plugins/dynamic-home-page
+    #backstage.io/source-location: file:../../plugins/dynamic-home-page/REQUIRED_TO_BUILD_THE_TECHDOCS_LOCALLY
+    backstage.io/view-url: https://github.com/janus-idp/backstage-showcase/blob/main/plugins/dynamic-home-page/catalog-info.yaml
+    backstage.io/edit-url: https://github.com/janus-idp/backstage-showcase/edit/main/plugins/dynamic-home-page/catalog-info.yaml
+    backstage.io/techdocs-ref: dir:.
+    github.com/project-slug: janus-idp/backstage-showcase
+    github.com/team-slug: janus-idp/rhtap
```
Comment (invincibleJai) [inline plugins/dynamic-home-page/catalog-info.yaml:15]: why rhtap? should this be rhdh?

File: plugins/dynamic-home-page/docs/layout-options.md [hunk 3]
```diff
@@ -0,0 +1,334 @@
+# Layout options
+
+The cards and content of the home page is rendered in a **12-column-grid**.
+
+All cards have a **fixed height** that can be changed by the admin.
+
+![Layout columns](layout-columns.png)
+
+<!--
```
Comment (invincibleJai) [inline plugins/dynamic-home-page/docs/layout-options.md:9]: does this needs to be commented?

File: plugins/dynamic-home-page/catalog-info.yaml (hunk 1 above)
Comment (jerolimov) [reply plugins/dynamic-home-page/catalog-info.yaml:38]: One is backstage-plugin (something we should show users)

```
  name: janus-idp-dynamic-home-page
  title: Dynamic Home Page plugin
```

The other one is the frontend-plugin:

```
  name: janus-idp-dynamic-home-page-frontend
  title: '@janus-idp/backstage-plugin-dynamic-home-page'
```

File: plugins/dynamic-home-page/catalog-info.yaml (hunk 2 above)
Comment (jerolimov) [reply plugins/dynamic-home-page/catalog-info.yaml:15]: Good catch, I changed it to `janus-idp/maintainers-plugins`, what we used in backstage-plugins as well.

File: plugins/dynamic-home-page/docs/layout-options.md (hunk 3 above)
Comment (jerolimov) [reply plugins/dynamic-home-page/docs/layout-options.md:9]: It was a long example for the screenshot. But you're right, there are better option to keep that example.

**Review by invincibleJai — APPROVED** [review]
/lgtm
/approve

**Pull request merged**